db/version_set.cc

```diff
@@ -2329,13 +2366,34 @@ void VersionStorageInfo::ComputeCompactionScore(
     } else {
       // Compute the ratio of current size to size limit.
       uint64_t level_bytes_no_compacting = 0;
+      uint64_t ingest_files_size = 0;
       for (auto f : files_[level]) {
         if (!f->being_compacted) {
-          level_bytes_no_compacting += f->compensated_file_size;
+          if (CanIgnoreFile(f, level)) {
+            ingest_files_size += f->compensated_file_size;
+          } else {
+            level_bytes_no_compacting += f->compensated_file_size;
+          }
         }
       }
       score = static_cast<double>(level_bytes_no_compacting) /
               MaxBytesForLevel(level);
+
+      if (level >= base_level_ && dynamic_level_bytes_ &&
+          level_bytes_no_compacting + ingest_files_size >
+              (1 +
+               static_cast<double>(mutable_cf_options.ingest_tolerant_ratio) /
+                   (level - base_level_ + 1)) *
+                  MaxBytesForLevel(level)) {
+        score = std::max(
+            score,
+            static_cast<double>(level_bytes_no_compacting + ingest_files_size) /
+                ((1 +
+                  static_cast<double>(
+                      mutable_cf_options.ingest_tolerant_ratio) /
```

Comment:
what would be the default `ingest_tolerant_ratio` used for tikv?

Reply:
I use 10 for test